### DISCUSSION TITLE
tri-state switch

### DISCUSSION THREAD

**Discussion by 123niel** [SchwarzIT/onyx#593] [Ideas]
It would be greate to have some sort of tri-state switch or toggle in onyx. Sometimes a simple boolean just isn't enough.

**Comment by mj-hof** [comment]
Hi Niel. Thank you very much for your input :). Could you describe your idea in more detail? Maybe you have an example of a use case/screenshot or something else for reference?

**Reply by 123niel** [reply]
Sure, in my view a tri-state switch has multiple use cases.
One example would be a filter form to filter entities that have multiple boolean properties. With tri-state switches you can enable user to set the filter to three values:

1. `neutral`: ignore this property
2. `positive`: only show entites where the property is true
3. `negative`: only show entities where the property is false

For our current Core UI applications we solved this with a dropdown select but I think something like a tri-state toggle would be more intuitive.

I have found some example implementations here: https://stackoverflow.com/questions/23661970/3-state-css-toggle-switch. I also took this screenshot from there:
![image](https://github.com/SchwarzIT/onyx/assets/24916146/73e3855b-70cf-481a-bcd7-2fe46389e7aa)

An intresting alternative might also be mutual exclusive toggle buttons like discussed here: https://ux.stackexchange.com/questions/80728/are-two-toggle-buttons-better-than-a-tri-state-switch

**Reply by JoCa96** [reply]
IMHO, the toggle button is the better approach, as a **switch** conveys the idea of a two-state input.

**Comment by larsrickert** [comment]
I like the idea of a tri-state switch. Another use case could also be a dark mode toggle where you can toggle between light, dark and automatic

**Comment by mj-hof** [comment]
@jannick-ux Please have a look at this

**Comment by jannick-ux** [comment]
Hi Niel, 
thanks for your input. We are still working on the basic component set for onyx and therefore there is a lot of thinks to come up in the future. In general we really like the overall functionality, and we already thought about similar things, but we have do do a clean conception and ux-testing to all the dependencies here first. 
So we created a [ticket](https://github.com/SchwarzIT/onyx/issues/753) to do a poc here. With this ticket, you can follow the status of your request. 

Thank you

**Comment by larsrickert** [comment]
@123niel We will provide a new component which looks like this:
<img width="408" alt="image" src="https://github.com/user-attachments/assets/f050603f-5676-4534-81e4-b3091985b557" />

You can track the progress in #2765
